packages/driver/cypress/integration/e2e/multi-domain/multi_domain_cypress_api.spec.ts

```diff
@@ -0,0 +1,286 @@
+  context.skip('session', () => {
+    it('clearAllSavedSessions() functions as expected', () => {
+      cy.switchToDomain('foobar.com', () => {
+        Cypress.session.clearAllSavedSessions()
+      })
+    })
+  })
+
+  context('properties', () => {
+    it('has arch property synced from primary', () => {
+      cy.switchToDomain('foobar.com', [Cypress.arch], ([theArch]) => {
+        expect(Cypress.arch).to.equal(theArch)
+      })
+    })
+
+    it('has browser property synced from primary', () => {
+      cy.switchToDomain('foobar.com', [Cypress.browser], ([theBrowser]) => {
+        expect(Cypress.browser).to.deep.equal(theBrowser)
+      })
+    })
+
+    it('has currentTest property synced from primary', () => {
+      cy.switchToDomain('foobar.com', [Cypress.currentTest], ([theCurrentTest]) => {
+        expect(Cypress.currentTest).to.deep.equal(theCurrentTest)
+      })
+    })
+
+    it('has platform property synced from primary', () => {
+      cy.switchToDomain('foobar.com', [Cypress.platform], ([thePlatform]) => {
+        expect(Cypress.platform).to.equal(thePlatform)
+      })
+    })
+
+    it('has testingType property synced from primary', () => {
+      cy.switchToDomain('foobar.com', [Cypress.testingType], ([theTestingType]) => {
+        expect(Cypress.testingType).to.deep.equal(theTestingType)
+      })
+    })
+
+    it('has spec property synced from primary', () => {
+      cy.switchToDomain('foobar.com', [Cypress.spec], ([theSpec]) => {
+        expect(Cypress.spec).to.deep.equal(theSpec)
+      })
+    })
+  })
+
+  context('methods', () => {
+    it('isCy()', () => {
+      cy.switchToDomain('foobar.com', () => {
+        expect(Cypress.isCy(cy)).to.be.true
+      })
+    })
+
+    it('isBrowser()', () => {
+      cy.switchToDomain('foobar.com', [Cypress.browser], ([theBrowser]) => {
+        expect(Cypress.isBrowser(theBrowser.name)).to.equal(true)
+      })
+    })
+
+    it('log()', (done) => {
+      cy.on('log:changed', (changedLog) => {
+        if (changedLog?.message === 'test log' && changedLog.ended) {
+          // just make sure Big Cypress logs make their way back to the primary
+          done()
+        }
+      })
+
+      cy.switchToDomain('foobar.com', () => {
+        Cypress.log({
+          message: 'test log',
+        })
+      })
+    })
+
+    // verifies the config from the primary domain is synced with the secondary
+    it('config()', () => {
+      const conf = Cypress.config()
+
+      cy.switchToDomain('foobar.com', [conf], ([theConfig]) => {
+        const multiDomainConfig = Cypress.config()
+
+        // this assertion takes a while to run... There is also some properties in the serialized config that differ, such as env: undefined
+        expect(theConfig).to.deep.include(multiDomainConfig)
+      })
+    })
+
+    /**
+     * Cypress.env() is not synced intentionally. Instead, maybe encourage users to pass in env variables via args to switchToDomain for security reasons?
```

Comment:
what are the thoughts on this?

Reply:
Since this would deviate from our inherit in pattern, we should probably talk about it in our next meeting.

Reply:
Agreed. I am a bit confused because I thought we decided 'own world' for everything else except `config` 😅 . Sounds like we definitely need to re-discuss this.

Reply:
Adding to the agenda